### PR TITLE
Setup Codeql workflow

### PR DESCRIPTION
**Pull request by mmorel-35**
**What type of PR is this?**
* chore: setup codeql workflow 

**What this PR does / why we need it**:
Provides a Static Application Security Testing to comply with OSSF rules.

**Which issue(s) this PR fixes**:
Relates to #2347

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Comment by mmorel-35** [comment]
I have rebased. This is available for merging when you are ready.

**Comment by mmorel-35** [comment]
cc @phlax

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  - cron: '16 11 * * 5'
+
+permissions:
+  contents: read
+
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-22.04' }}
```
Comment (arkodg) [inline .github/workflows/codeql.yml:20]: why is `swift` and `macos-latest` needed here ?

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**